types/src/transaction/mod.rs

```diff
@@ -737,7 +738,41 @@ impl TransactionWithProof {
         self.proof.verify(ledger_info, version)
     }
 }
+/// The status of VM execution, which contains more detailed failure info
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub enum KeptTXNStatus {
+    Executed,
+    OutOfGas,
+    MoveAbort(AbortLocation, /* code */ u64),
```

Comment:
why not give fields names?
```
MoveAbort {location: AbortLocation, code: u64}

Reply:
And not possible to create an enum to list all the codes possible?

Reply:
it has to be type code: u64 won't compile.  Not very doable to list as a enum https://github.com/areshand/aptos-core-1/blob/1186a8ade10acdbb8b69886a7331492603aee301/sdk/transaction-builder/src/error_explain.rs#L23 is currently for explaining all the code.